webui/src/components/Jobs.jsx

```diff
@@ -147,13 +161,12 @@ class Jobs extends React.Component {
                 (service && service.ID && service.ID.length) ? service.ID : "unknown service";
             let title = "Job from " + serviceName;
 
-            let execDuration = "";
-            if (job.State.Code == -1) {
-                let currentDate = new Date();
-                execDuration = currentDate - Date.parse(job.Created);
+            let jobStartTime = new Date(job.Created);
+            let jobDurationTime = new Date(job.Duration);
+            let jobFinishTime = new Date(jobStartTime.getTime() + jobDurationTime.getTime());
```

Comment:
This seems incorrect, server duration is in seconds and new Date accepts milliseconds. You should use:
`let jobFinishTime = new Date(jobStartTime.getTime() + 1000 * job.Duration);`